### DISCUSSION TITLE
Is server components only available in the/app folder? Can I use it under pages?

### DISCUSSION THREAD

**Discussion by Lrunlin** [vercel/next.js#44878] [Help]
![image](https://user-images.githubusercontent.com/54397964/212449068-547b2165-1699-45d2-8cb4-2cd097c34167.png)

**Comment by icyJoseph** [comment]
I think so yes. The idea being that Next.js doesn't want to prevent people from continuously working on their current `pages` based projects. The appDir and server components require a bunch of changes that are just not practical to lay on top of `pages`.

**Reply by Lrunlin** [reply]
Thank you. I have learned the basic upgrade rules and the use of/app through the Next.js document
